tests/document_ops.rs

```diff
@@ -72,6 +74,68 @@ fn doc_create_with_grant() {
         ])
     )
 }
+
+#[test]
+fn doc_create_without_self_grant() {
+    let mut sdk = init_sdk();
+
+    let doc = [0u8; 64];
+
+    // create a second user to grant access to the document
+    let second_user = create_second_user();
+
+    let doc_result = sdk
+        .document_encrypt(
+            &doc,
+            &DocumentEncryptOpts::new(
+                None,
+                Some("first name".try_into().unwrap()),
+                false,
+                vec![UserOrGroup::User {
+                    id: second_user.account_id().clone(),
+                }],
+            ),
+        )
+        .unwrap();
+
+    // should be a user with access, but not the currently initd user
+    assert_eq!(doc_result.grants().len(), 1);
+    assert_ne!(
+        doc_result.grants()[0],
+        UserOrGroup::User {
+            id: sdk.device().account_id().clone()
+        }
+    );
+    assert_eq!(
+        doc_result.grants()[0],
+        UserOrGroup::User {
+            id: second_user.account_id().clone()
+        }
+    );
+    assert_eq!(doc_result.access_errs().len(), 0);
+}
+
+#[test]
+fn doc_create_must_grant() {
+    let mut sdk = init_sdk();
+
+    let doc = [0u8; 64];
+
+    // should fail because encrypting a document with no grants is nonsense
+    let doc_result = sdk.document_encrypt(
+        &doc,
+        &DocumentEncryptOpts::new(None, Some("first name".try_into().unwrap()), false, vec![]),
+    );
+
+    assert_eq!(
+        doc_result.err().unwrap(),
+        IronOxideErr::ValidationError(
+            "grant_to_author".to_string(),
+            "grant_to_author cannot be false if there are no explicit grants".to_string()
```

Comment:
I'm not a huge fan of duplicating this exact string in the unit tests. I think it'd be good to have some verification (that it contains "grant_to_author" or similar). But I don't think we care enough about this specific error string that if we decide to change it the unit tests should then start failing.

Reply:
Is this better?

```rs
    // make sure there was a validation error, and that the problem was with the grant
    assert_eq!(
        match doc_result.err().unwrap() {
            IronOxideErr::ValidationError(field_name, _) => field_name,
            _ => "failed test".to_string(),
        },
        "grant_to_author".to_string()
    )
```

Reply:
👍